pkg/kubectl/cmd/delete_flags.go

```diff
@@ -72,9 +73,10 @@ type DeleteFlags struct {
 	Output         *string
 }
 
-func (f *DeleteFlags) ToOptions(streams genericclioptions.IOStreams) *DeleteOptions {
+func (f *DeleteFlags) ToOptions(dynamicClient dynamic.Interface, streams genericclioptions.IOStreams) *DeleteOptions {
```

Comment:
So passing `dynamicClient` here is really only necessary for the other commands that use deletion (`apply`, `replace`), normal `delete` command will initialize its `DynamicClient` field in the `Complete()` method...

Reply:
Yes, but we're doing this so that other places in the code re-use and not create their own versions of wait.